src/jit/codegencommon.cpp

```diff
@@ -7369,10 +7398,18 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // Can't have a call until we have enough padding for rejit
     genPrologPadForReJit();
 
+#if !defined(UNIX_AMD64_ABI)
     // This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov rax, helper addr; call rax"
     genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN);
+#else
+    // We can use any callee trash register (other than RAX, RDI, RSI) for call target.
+    // We use R11 here. This will emit either
+    // "call ip-relative 32-bit offset" or
+    // "mov r11, helper addr; call r11"
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_R11);
```

Comment:
Then we should disable the arg reg homing logic for unix amd64.  The fact this logic was not disabled confused me.

Next, R12 and R13 are callee saved regs. They have to be saved in method prolog and restored in epilog.  Another point I am trying to convey is that just adding them to mask vars is not enough to get them saved/restored.